Security/src/CloudFoundrySingleSignon/Startup.cs

```diff
@@ -33,12 +33,15 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
 
             services.AddCloudFoundryContainerIdentity();
-            services.AddHttpClient("default", (services, client) =>
+            services.AddHttpClient("default", (serviceProvider, client) =>
             {
-                var options = services.GetService<IOptions<CertificateOptions>>();
-                var b64 = Convert.ToBase64String(options.Value.Certificate.Export(X509ContentType.Cert));
-                client.DefaultRequestHeaders.Add("X-Forwarded-Client-Cert", b64);
-            }).ConfigurePrimaryHttpMessageHandler((isp) => new HttpClientHandler { ServerCertificateCustomValidationCallback = (sender, cert, chain, sslPolicyErrors) => true });
+                var options = serviceProvider.GetService<IOptions<CertificateOptions>>();
+                if (options?.Value.Certificate != null)
```

Comment:
I don't think `GetService` for an `IOptions<>` ever returns null. If there's no configuration, you'll get a default instance.

```suggestion
                var options = serviceProvider.GetRequiredService<IOptions<CertificateOptions>>();
                if (options.Value.Certificate != null)
```

The same suggestion applies to other places in this PR, which I haven't marked.